File: rock63/src/main/java/com/uksusoff/rock63/jobs/DataUpdateJob.java
package com.uksusoff.rock63.jobs;

import androidx.annotation.NonNull;

import com.bugsnag.android.Bugsnag;
import com.evernote.android.job.Job;
import com.uksusoff.rock63.data.DataSource;
import com.uksusoff.rock63.exceptions.NoContentException;
import com.uksusoff.rock63.exceptions.NoInternetException;

import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.EBean;

@EBean()
public class DataUpdateJob extends Job {

    public static final String TAG = "data_update_job";

    @Bean
    DataSource source;

    @NonNull
    @Override
    protected Result onRunJob(Params params) {
        try {
            source.sourcesRefresh();
        } catch (NoInternetException | NoContentException e) {
            //Just not this time
            Bugsnag.notify(e);
        } catch (Exception e) {
            Bugsnag.notify(e);
        }

        return Result.SUCCESS;
    }

}

File: rock63/src/main/java/com/uksusoff/rock63/jobs/DataUpdateJob.kt
package com.uksusoff.rock63.jobs

import com.evernote.android.job.Job
import com.uksusoff.rock63.data.DataProviderComponent
import com.uksusoff.rock63.data.DataProviderComponent.NoInternetException
import com.uksusoff.rock63.data.DataProviderComponent_

/**
 * Created by User on 28.08.2016.
 */
class DataUpdateJob : Job() {
    override fun onRunJob(params: Params): Result {
        val dataProviderComponent: DataProviderComponent = DataProviderComponent_.getInstance_(context)
        try {
            dataProviderComponent.refreshEvents()
            dataProviderComponent.refreshNews()
        } catch (e: NoInternetException) { //Just not this time
        }
        return Result.SUCCESS
    }

    companion object {
        const val TAG = "data_update_job"
    }
}